src/utils/options.py

```diff
@@ -0,0 +1,313 @@
+        restricted[:] = [line for line in restricted if len(line.split()) != 0]  # remove blanks
+
+        for line in restricted:
+            splitup = line.split("=")
+            if len(splitup) == 2:
+                if splitup[0] == 'default' or splitup[0] == 'default ':
+                    self.default_freq = int(splitup[1])  # kHz
+                    restricted.remove(line)
+                    break
+        else:  # no break
+            raise Exception('No Default Frequency Found in Restrict.dat')
+
+        self.restricted_ranges = []
+        for line in restricted:
+            splitup = line.split()
+            if len(splitup) != 2:
+                raise Exception('Problem with Restricted Frequency: A Range Len != 2')
+            try:
+                splitup = [int(float(freq)) for freq in splitup]  # convert to ints
+            except ValueError:
+                raise ValueError('Error parsing Restrict.Dat Frequency Ranges, Invalid Literal')
+            restricted_range = tuple(splitup)
+            self.restricted_ranges.append(restricted_range)
+
+    def verify_options(self):
+        if self.site_id != os.environ["RADAR_ID"]:
+            errmsg = f'site_id {self.site_id} is different from RADAR_ID {os.environ["RADAR_ID"]}'
+            raise ValueError(errmsg)
+
+        if len(self.n200_addrs) != len(set(self.n200_addrs)):
+            errmsg = 'Two or more n200s have identical IP addresses'
+            raise ValueError(errmsg)
+
+        if len(self.main_antennas) > 0:
+            if min(self.main_antennas) < 0 or max(self.main_antennas) >= self.main_antenna_count:
+                errmsg = 'main_antennas and main_antenna_count are not consistent'
+                raise ValueError(errmsg)
+        if len(self.intf_antennas) > 0:
+            if min(self.intf_antennas) < 0 or \
+                    max(self.intf_antennas) >= self.intf_antenna_count:
+                errmsg = 'intf_antennas and intf_antenna_count are not consistent'
+                raise ValueError(errmsg)
+        if len(self.main_antennas) != len(set(self.main_antennas)):
+            errmsg = 'main_antennas contains duplicate values'
+            raise ValueError(errmsg)
+        if len(self.intf_antennas) != len(set(self.intf_antennas)):
+            errmsg = 'intf_antennas contains duplicate values'
+            raise ValueError(errmsg)
+
+        # TODO: Test that realtime_address and router_address are valid addresses
+
+        if not os.path.exists(self.data_directory):
+            errmsg = f'data_directory {self.data_directory} does not exist'
+            raise ValueError(errmsg)
+        if not os.path.exists(self.log_directory):
+            errmsg = f'log_directory {self.log_directory} does not exist'
+            raise ValueError(errmsg)
+        if not os.path.exists(self.hdw_path):
+            errmsg = f'hdw_path directory {self.hdw_path} does not exist'
+            raise ValueError(errmsg)
+
+    def __repr__(self):
```

Comment:
dataclasses generate their own __repr__ function (https://docs.python.org/3.8/library/dataclasses.html#, arguments to the `@dataclass` decorator) so I think we can remove this method. We never print out the Options class anyways

Reply:
Changed the name from `__repr__`  to `__str__` to leave the built in `repr` do what it does